client/src/pages/UserPage.tsx

```diff
@@ -0,0 +1,276 @@
+import { useEffect, useRef, useState } from 'react';
+import './UserPage.css';
+import { ProgressReport } from '../components/ProgressAssessment';
+import { ProgressAssessment } from '../components/ProgressAssessment';
+import { Modal } from '../components/Modal';
+import { Filler } from 'chart.js';
+import { Line } from 'react-chartjs-2';
+import {
+  Chart as ChartJS,
+  CategoryScale,
+  LinearScale,
+  PointElement,
+  LineElement,
+  Title,
+  Tooltip,
+  Legend,
+} from 'chart.js';
+import { Link, useNavigate } from 'react-router-dom';
+import { readToken } from '../lib';
+import { useUser } from '../components/useUser';
+
+ChartJS.register(
+  CategoryScale,
+  LinearScale,
+  PointElement,
+  LineElement,
+  Title,
+  Tooltip,
+  Legend,
+  Filler
+);
+
+export type User = {
+  userId?: number;
+  userName: string;
+};
+
+export function UserPage() {
+  const [isOpen, setIsOpen] = useState(false);
+  const [scoreHistory, setScoreHistory] = useState<ProgressReport[]>([]);
+
+  const modal = useRef<HTMLDialogElement>(null);
+  const bear = readToken();
+
+  const { handleSignOut, user } = useUser();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    async function getData() {
+      try {
+        const response = await fetch('/api/progressassessment', {
+          method: 'GET',
+          headers: {
+            'Content-Type': 'application/json',
+            Authorization: `Bearer ${bear}`,
+          },
+        });
+        if (!response.ok) {
+          throw new Error(`Response status: ${response.status}`);
+        }
+
+        const json = (await response.json()) as ProgressReport[];
+        setScoreHistory(json);
+      } catch (error) {
+        console.error('Error fetching progress score:', error);
+      }
+    }
+    if (user) {
+      getData();
+    }
+  }, [bear, user]);
+
+  function handleSuccess(responseData: ProgressReport) {
+    setScoreHistory([...scoreHistory, responseData]);
+    setIsOpen(false);
+  }
+
+  function openModal() {
+    modal.current?.showModal();
```

Comment:
This looks like it's leftover from before you used the Modal component. Remove it, and the `modal` useRef call. Same with line 84.